scripts/convert_ima_policy.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import copy
+import datetime
+import json
+import re
+import sys
+from os.path import basename
+
+from keylime.ima import file_signatures, ima
+
+"""
+This script converts legacy Keylime policies into the unified, next-generation
+IMA policy format.Both legacy allow/exclude lists as well as out-of-date IMA
+policy formats are accepted.
+
+Example usage:
+
+To convert an allowlist and exclude list to an IMA policy at `keylime-policy.json`:
+
+```
+python convert_ima_policy.py -a <allowlist path> -e <exclude list path> -o keylime-policy.json
+```
+
+To convert an existing IMA policy to the latest version:
+
+```
+python convert_ima_policy.py -i <ima policy path> -o keylime-policy.json
+```
+
+To view help and see all available options:
+
+```
+python convert_ima_policy.py -h
+```
+"""
+
+# Creates an IMA policy from provided legacy allowlist.
+def create_ima_policy(allowlist_path):
+    with open(allowlist_path, "r") as f:
+        alist_raw = f.read()
+
+    p = re.compile(r"^\s*{")
+    if p.match(alist_raw):
+        print(f"{basename(allowlist_path)} is JSON-formatted; converting to IMA policy")
+        alist_json = json.loads(alist_raw)
```

Comment:
Should do this in a try/catch block because it's user input and will fail in all kinds of weird ways. Best to catch it and give a friendlier error message.